test/acceptance/test/profiles_test.go

```diff
@@ -0,0 +1,171 @@
+		Expect(statusCode).To(Equal(http.StatusOK))
+
+		profiles := pb.GetProfilesResponse{}
+		Expect(json.Unmarshal(resp, &profiles)).To(Succeed())
+		Expect(profiles.Profiles).NotTo(BeNil())
+		Expect(profiles.Profiles).To(ConsistOf(&pb.Profile{
+			Name:        "podinfo",
+			Home:        "https://github.com/stefanprodan/podinfo",
+			Description: "Podinfo Helm chart for Kubernetes",
+			Sources:     []string{"https://github.com/stefanprodan/podinfo"},
+			Maintainers: []*pb.Maintainer{
+				{
+					Name:  "stefanprodan",
+					Email: "stefanprodan@users.noreply.github.com",
+				},
+			},
+			//These have to not be nil for the assertion to work. because proto :shrug:
+			Keywords:    []string{},
+			Annotations: map[string]string{},
+		}))
+
+		By("Getting the values for a profile")
+		resp, statusCode, err = kubernetesDoRequest(namespace, profilesSvc, "8000", "/v1/profiles/podinfo/6.0.1/values", clientSet)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(statusCode).To(Equal(http.StatusOK))
+
+		profileValues := pb.GetProfileValuesResponse{}
+		Expect(json.Unmarshal(resp, &profileValues)).To(Succeed())
+		values, err := base64.StdEncoding.DecodeString(profileValues.Values)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(string(values)).To(ContainSubstring("# Default values for podinfo"))
+	})
+})
+
+func buildKubernetesClients() (*kubernetes.Clientset, client.Client) {
+	config, err := clientcmd.BuildConfigFromFlags("", filepath.Join(homedir.HomeDir(), ".kube", "config"))
+	Expect(err).NotTo(HaveOccurred())
+	clientSet, err := kubernetes.NewForConfig(config)
+	Expect(err).NotTo(HaveOccurred())
+
+	scheme := runtime.NewScheme()
+	schemeBuilder := runtime.SchemeBuilder{
+		sourcev1beta1.AddToScheme,
+	}
+	Expect(schemeBuilder.AddToScheme(scheme)).To(Succeed())
+
+	kClient, err := client.New(config, client.Options{
+		Scheme: scheme,
+	})
+
+	Expect(err).NotTo(HaveOccurred())
+
+	return clientSet, kClient
+}
+
+func deployProfilesHelmRepository(rawClient client.Client, namespace string) {
+	weaveworksChartsHelmRepository := &sourcev1beta1.HelmRepository{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       sourcev1beta1.HelmRepositoryKind,
+			APIVersion: sourcev1beta1.GroupVersion.Identifier(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "weaveworks-charts",
+			Namespace: namespace,
+		},
+		Spec: sourcev1beta1.HelmRepositorySpec{
+			URL:      "https://weaveworks.github.io/profiles-examples",
+			Interval: metav1.Duration{Duration: time.Minute * 10},
+		},
+	}
+	err = rawClient.Create(context.TODO(), weaveworksChartsHelmRepository)
+	Expect(err).NotTo(HaveOccurred())
+	Eventually(func() error {
+		helmRepo := sourcev1beta1.HelmRepository{}
+		err := rawClient.Get(context.TODO(), client.ObjectKeyFromObject(weaveworksChartsHelmRepository), &helmRepo)
+		if err != nil {
+			return err
+		}
+
+		if len(helmRepo.Status.Conditions) == 1 && helmRepo.Status.Conditions[0].Type == "Ready" && helmRepo.Status.Conditions[0].Status == "True" {
```

Comment:
This check is very prone to errors due to minimal changes in contract (e.g. the introduction of other conditions), and it would be much more resilient if `apimeta.FindStatusCondition` would be used: https://github.com/kubernetes/apimachinery/blob/master/pkg/api/meta/conditions.go#L74

Reply:
I like this! Will update